overlays/nodejs/s3/s3Mixins.ts

```diff
@@ -0,0 +1,231 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi"
+import { Bucket } from "./bucket";
+import { BucketNotification } from "./bucketNotification";
+import * as lambda from "../lambda"
+
+/**
+ * Arguments to help customize a notification subscription for a bucket.
+ */
+export interface CommonBucketSubscriptionArgs {
+    /**
+     * An optional prefix or suffix to filter down notifications.  See
+     * aws.s3.BucketNotification.lambdaFunctions for more details.
+     */
+    filterPrefix?: string;
+    filterSuffix?: string;
+}
```

Comment:
All this code moved down from aws-serverless.  However, it was tweaked to point at aws.lambda, not aws.serverless at this layer.